lib/metadata.ts

```diff
@@ -94,6 +98,22 @@ const parseMetadata = async (idpMeta: string, validateOpts): Promise<Record<stri
 
         if (X509Certificate) {
           ret.thumbprint = thumbprint.calculate(X509Certificate);
+          /**
+           * new crypto.X509Certificate fails with the X509Certificate cert without
+           * -----BEGIN CERTIFICATE-----
+           * and
+           * -----END CERTIFICATE-----
+           */
+          if (X509Certificate.indexOf(BEGIN) != -1 && X509Certificate.indexOf(END) != -1) {
```

Comment:
Shouldn't the `else` block be the one that should execute here ?

Reply:
@niwsa The if block is looking for missing begin and end blocks

Reply:
Got it.